Sources/Network/URLSessionNetworkStack.swift

```diff
@@ -10,36 +10,48 @@ import Foundation
 
 public extension Network {
 
-    final class URLSessionNetworkStack: NSObject, NetworkStack, URLSessionDelegate {
+    public final class CancelableTask: Cancelable {
+        private weak var task: URLSessionTask?
 
-        private struct CancelableTask: Cancelable {
-            private weak var task: URLSessionTask?
+        init(task: URLSessionTask) {
+            self.task = task
+        }
 
-            init(task: URLSessionTask? = nil) {
-                self.task = task
-            }
+        public func cancel() {
+            task?.cancel()
+        }
+    }
 
-            mutating func set(task: URLSessionTask) {
-                self.task = task
-            }
+    public final class CancelableBag: Cancelable {
+        private lazy var cancelables: [Cancelable] = []
 
-            public func cancel() {
-                task?.cancel()
-            }
+        public init() {}
+
+        deinit {
+            cancel()
```

Comment:
not sure about this. I might want to discard the `Cancelable`, which may cause it do be `deinit`ed when I simply want the task to complete